### ISSUE TITLE
Is it possible the directoryiterator or flow_from_directory takes multiple directories as input ?

### ISSUE THREAD

**Issue by easilar** [keras-team/keras-preprocessing#67]
Dear Keras team,
I need to combine data from multiple directories with exactly same sub directory structure. 
Would it be possible to add this feature into flow_from_directory class ?

With my restricted programming knowledge I would simply modify/add some lines before the line below. 
https://github.com/keras-team/keras-preprocessing/blob/2c7ef1d27bff4a24fe8bd2dc3800ccdebfc08a75/keras_preprocessing/image.py#L1888
?

**Comment by Dref360** [comment]
This is a really particular usecase.
I think you should use `flow_from_dataframe` to solve your issue.

**Comment by easilar** [comment]
Yes I can put all my data to dataframe however then flow_from_dataframe needs the directory also as an argument. If I could give a directory I would use the flow_from_directory. I don't understand why flow from dataframe needs directory. 
I created my dataframe as here: 
for train_region in training_regions:
      X = numpy array from train_region
      pd_dataset = pd.DataFrame({'label': Y, 'images': list(X)}, columns=['label', 'images'])
      pd_datasets.append(pd_dataset)
data_train = pd.concat(pd_datasets)
The problem is I can not have one directory for all training images ...

**Comment by qinst64** [comment]
#56 
Don't understand either why it needs the `directory`  argument and `x_col ` needs to be filenames only (not full path) otherwise it raises error

**Comment by rragundez** [comment]
This has been solved in the current version. This issue can be closed @Dref360
